src/Ztm.Zcoin.Rpc/ZcoinRpcClient.cs

```diff
@@ -445,6 +445,11 @@ async Task<ExodusTransaction> TryDecodeExodusTransaction(Transaction transaction
 
         async Task TryDecodeExodusTransactionAndSetAsync(Transaction transaction)
         {
+            if (transaction.GetHash() == this.client.Network.GetGenesis().Transactions.First().GetHash())
```

Comment:
We should cache the hash of transactions that contains in the genesis block instead due to `GetGenesis` will always deserialize block from hex string.